### PR TITLE
Add peer_region to vpc peering connection

### PR DESCRIPTION
**Pull request by kl4w**
https://github.com/terraform-providers/terraform-provider-aws/issues/2484

```bash
TF_ACC=1 go test ./aws -v -run=TestAccAWSVPCPeeringConnection_region -timeout 120m                                                                                                                                     ~/D/k/r/g/s/g/t/terraform-provider-aws inter-region-vpc-peer
=== RUN   TestAccAWSVPCPeeringConnection_region
--- PASS: TestAccAWSVPCPeeringConnection_region (27.57s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	27.615s
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
[`aws_vpc_peering_connection` data source](https://www.terraform.io/docs/providers/aws/d/vpc_peering_connection.html) and [`aws_vpc_peering_connection_accepter` resource](https://www.terraform.io/docs/providers/aws/r/vpc_peering_accepter.html) also need updating.

**Comment by kl4w** [comment]
Should I do it as part of this PR, or separate it?

**Comment by ewbankkit** [comment]
Seems like a pretty small change so I'd say do it as part of this PR.

**Comment by kl4w** [comment]
rebased and found an issue (not exactly sure why the test before didn't pick this up).

`aws_vpc_peering_connection.foo: Unable to accept VPC Peering Connection: OperationNotPermitted: Incorrect region (us-west-2) specified for this request. VPC peering connection pcx-029d90c5cc8823141 must be accepted in region us-east-1`

I'm having a dangling resource issue in the test. I'm wondering whether it may be worth it to note that if a peer connection is in a different region/account, the `aws_vpc_peering_accepter` resource must be used

**Comment by kl4w** [comment]
Found a couple of issues and would like some input for the below errors:

```
TF_ACC=1 go test ./aws -v -run=TestAccAWSVPCPeeringConnection -timeout 120m
=== RUN   TestAccAWSVPCPeeringConnection_importBasic
--- PASS: TestAccAWSVPCPeeringConnection_importBasic (68.59s)
=== RUN   TestAccAWSVPCPeeringConnectionAccepter_sameRegion
--- PASS: TestAccAWSVPCPeeringConnectionAccepter_sameRegion (45.91s)
=== RUN   TestAccAWSVPCPeeringConnectionAccepter_differentRegion
--- FAIL: TestAccAWSVPCPeeringConnectionAccepter_differentRegion (25.63s)
	testing.go:503: Step 0 error: Error applying: 1 error(s) occurred:

		* aws_vpc_peering_connection_accepter.peer: 1 error(s) occurred:

		* aws_vpc_peering_connection_accepter.peer: Unable to accept VPC Peering Connection: OperationNotPermitted: Incorrect region (us-west-2) specified for this request. VPC peering connection pcx-0e1b4b8ba14e6e2a9 must be accepted in region us-east-1
			status code: 400, request id: fe232759-0bb6-4987-8b04-5dfa8857b60d
=== RUN   TestAccAWSVPCPeeringConnection_basic
--- PASS: TestAccAWSVPCPeeringConnection_basic (38.28s)
=== RUN   TestAccAWSVPCPeeringConnection_plan
--- PASS: TestAccAWSVPCPeeringConnection_plan (34.98s)
=== RUN   TestAccAWSVPCPeeringConnection_tags
--- PASS: TestAccAWSVPCPeeringConnection_tags (40.90s)
=== RUN   TestAccAWSVPCPeeringConnection_options
--- PASS: TestAccAWSVPCPeeringConnection_options (64.33s)
=== RUN   TestAccAWSVPCPeeringConnection_failedState
--- PASS: TestAccAWSVPCPeeringConnection_failedState (24.87s)
=== RUN   TestAccAWSVPCPeeringConnection_peerRegionAndAutoAccept
--- PASS: TestAccAWSVPCPeeringConnection_peerRegionAndAutoAccept (21.43s)
=== RUN   TestAccAWSVPCPeeringConnection_region
--- FAIL: TestAccAWSVPCPeeringConnection_region (37.76s)
	testing.go:563: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Check failed: Found the VPC Peering Connection in an unexpected state: {
		  AccepterVpcInfo: {
		    OwnerId: "blah",
		    Region: "us-east-1",
		    VpcId: "vpc-7b6d0703"
		  },
		  RequesterVpcInfo: {
		    OwnerId: "blah",
		    Region: "us-west-2",
		    VpcId: "vpc-bdb1afdb"
		  },
		  Status: {
		    Code: "rejected",
		    Message: "Rejected by blah"
		  },
		  VpcPeeringConnectionId: "pcx-0b0721f1be30fd4c0"
		}

		State: <no state>
FAIL
exit status 1
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	402.734s
```

`TF_LOG=DEBUG TF_ACC=1 go test ./aws -v -run=TestAccAWSVPCPeeringConnection_region -timeout 120m`
The dangling resource seems to happen when attempting to delete `aws_vpc.bar`. The request looks like it's going to `eu-west-2` rather than `us-east-1`
```
-----------------------------------------------------
2017/12/03 09:34:47 [DEBUG] [aws-sdk-go] DEBUG: Request ec2/DeleteVpc Details:
---[ REQUEST POST-SIGN ]-----------------------------
POST / HTTP/1.1
Host: ec2.us-west-2.amazonaws.com
User-Agent: aws-sdk-go/1.12.38 (go1.9.2; darwin; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.11.0-beta1
Authorization: AWS4-HMAC-SHA256 Credential=blah/20171203/us-west-2/ec2/aws4_request
```

`TF_LOG=DEBUG TF_ACC=1 go test ./aws -v -run=TestAccAWSVPCPeeringConnectionAccepter_differentRegion -timeout 120m`

when attempting to call accept `aws_vpc_peering_connection_accepter.peer` the request seems to be going to `eu-west-2` rather than `us-east-1`.
```
2017/12/03 09:45:14 [DEBUG] [aws-sdk-go] DEBUG: Request ec2/AcceptVpcPeeringConnection Details:
---[ REQUEST POST-SIGN ]-----------------------------
POST / HTTP/1.1
Host: ec2.us-west-2.amazonaws.com
User-Agent: aws-sdk-go/1.12.38 (go1.9.2; darwin; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.11.0-beta1
Authorization: AWS4-HMAC-SHA256 Credential=blah/20171203/us-west-2/ec2/aws4_request
```

**Comment by kl4w** [comment]
TF_ACC=1 go test ./aws -v -run=TestAccAWSVPCPeeringConnection -timeout 120m                                                                                                                                         ~/D/k/r/g/s/g/t/terraform-provider-aws inter-region-vpc-peer
=== RUN   TestAccAWSVPCPeeringConnection_importBasic
--- PASS: TestAccAWSVPCPeeringConnection_importBasic (25.47s)
=== RUN   TestAccAWSVPCPeeringConnectionAccepter_sameRegion
--- PASS: TestAccAWSVPCPeeringConnectionAccepter_sameRegion (25.34s)
=== RUN   TestAccAWSVPCPeeringConnectionAccepter_differentRegion
--- PASS: TestAccAWSVPCPeeringConnectionAccepter_differentRegion (31.64s)
=== RUN   TestAccAWSVPCPeeringConnection_basic
--- PASS: TestAccAWSVPCPeeringConnection_basic (23.71s)
=== RUN   TestAccAWSVPCPeeringConnection_plan
--- PASS: TestAccAWSVPCPeeringConnection_plan (22.57s)
=== RUN   TestAccAWSVPCPeeringConnection_tags
--- PASS: TestAccAWSVPCPeeringConnection_tags (25.69s)
=== RUN   TestAccAWSVPCPeeringConnection_options
--- PASS: TestAccAWSVPCPeeringConnection_options (40.32s)
=== RUN   TestAccAWSVPCPeeringConnection_failedState
--- PASS: TestAccAWSVPCPeeringConnection_failedState (15.18s)
=== RUN   TestAccAWSVPCPeeringConnection_peerRegionAndAutoAccept
--- PASS: TestAccAWSVPCPeeringConnection_peerRegionAndAutoAccept (12.69s)
=== RUN   TestAccAWSVPCPeeringConnection_region
--- PASS: TestAccAWSVPCPeeringConnection_region (26.77s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	249.420s

**Comment by scrothers** [comment]
Is this going to make it in soon? Or is there documentation somewhere on how/when PRs get merged by Hashicorp?

**Comment by jmvbxx** [comment]
Is there an update on this issue? Any idea when it's going to be merged?

File: aws/resource_aws_vpc_peering_connection.go [hunk 1]
```diff
@@ -69,6 +75,13 @@ func resourceAwsVPCPeeringCreate(d *schema.ResourceData, meta interface{}) error
 		createOpts.PeerOwnerId = aws.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("peer_region"); ok {
+		if _, ok := d.GetOk("auto_accept"); ok {
+			return fmt.Errorf("peer_region cannot be set whilst auto_accept is true when creating a vpc peering connection")
+		}
+		createOpts.SetPeerRegion(v.(string))
```
Comment (ewbankkit) [inline aws/resource_aws_vpc_peering_connection.go]: The convention has been to address the structure's field directly, i.e.
`createOpts.PeerRegion = aws.String(v.(string))`

Comment (kl4w) [reply aws/resource_aws_vpc_peering_connection.go]: @ewbankkit Missed that, thanks!

About to push the change up, should I rebase and recomment the test results as well?

**Comment by erikbor** [comment]
Any updates on when this will be merged? Thanks!

**Comment by kl4w** [comment]
@ewbankkit how is this PR looking? Is this ready to be merged?

**Comment by ewbankkit** [comment]
I have no further comments but it is not my call to make.

**Comment by scrothers** [comment]
@radeksimko You seem to be the one doing some of the most recent merges in PRs. Can you please provide some insight into how PRs are merged and when?

It would help us understand the process so we don't seem so noisy or bothersome :)

**Comment by jen20** [comment]
This looks good to me, and the tests pass. There are a few minor things which I will fix up in a supplementary commit locally before merging.

However, it turns out that the resources for the accept*or* are misspelled accept*e*r - we should address that before the next release by adding aliases for the correct naming, and leave the current one for backward compatibility.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**